src/FufillerLogo.jsx

```diff
@@ -1,44 +1,99 @@
 import React, { Component } from 'react';
 import VisibilitySensor from 'react-visibility-sensor';
+import PropTypes from 'prop-types';
 
 export default class FulfillerLogo extends React.Component {
 
-  constructor(props) {
+
+  constructor (props) {
     super(props);
-    this.url = `https://fulfilleridentity.trdlnk.cimpress.io/v1/fulfillers/${props.fulfillerId}/logo`;
     this.state = {
       imageBlob: null,
+      fulfillerId: props.fulfillerId,
+      visible: false,
+      imageIsLoading: false,
+      imageIsForbidden: false,
+    };
+    this.defaultStyle = {
+      maxWidth: "50px",
+      maxHeight: "50px",
+      display: "inline-block",
+      objectFit: "contain",
+      width: "50px",
+      height: "50px",
+      marginRight: "10px",
+      verticalAlign: "middle"
+    };
+  }
+
+  componentWillReceiveProps (newProps) {
+    if (newProps.fulfillerId !== this.props.fulfillerId) {
+      this.setState({
+        imageBlob: null,
+        fulfillerId: newProps.fulfillerId,
+      }, () => this.fetchImage(this.state.visible));
     }
   }
 
-  fetchImage(isVisible) {
+  fetchImage (isVisible) {
+    this.setState({
+      visible: isVisible
+    });
     if (isVisible && !this.state.imageBlob) {
+      this.setState({
+        imageIsLoading: true
+      });
       let headers = new Headers();
-      headers.append("Authorization", `Bearer ${this.props.accessToken}`);
-      headers.append("Accept", "image/*");
-      let init = { method: 'GET',
+      headers.append('Authorization', `Bearer ${this.props.accessToken}`);
+      headers.append('Accept', 'image/*');
+      let init = {
+        method: 'GET',
         headers: headers,
         mode: 'cors',
-        cache: 'default' };
-      fetch(this.url, init).then(response => {
+        cache: 'default'
+      };
+      let url = `https://fulfilleridentity.trdlnk.cimpress.io/v1/fulfillers/${this.state.fulfillerId}/logo`;
```

Comment:
It is my understanding that, in React, this.props is accessible from within the object. Therefore, you do not need to retrieve fulfillerId from the state - and there's no reason to store them in the state at line 12.